hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -480,8 +482,20 @@ public Set<ContainerReplica> replicasToCopyToFixMisreplication(
       requiredNumberOfPlacementGroups -= 1;
       if (numberOfReplicasToBeCopied > 0) {
         List<ContainerReplica> replicasToBeCopied = replicaList.stream()
+                .filter(replicas::get)
                 .limit(numberOfReplicasToBeCopied)
                 .collect(Collectors.toList());
+        if (numberOfReplicasToBeCopied > replicasToBeCopied.size()) {
+          Node rack = this.getPlacementGroup(
```

Comment:
NIT: 

Could this be: `getPlacemenGroup(replicaList.get(0).getDatanodeDetails())` ?

Reply:
done